App.js

```diff
@@ -161,73 +322,311 @@ export default function App() {
         workMessage = 'Woops...roll set to 0';
       }
       // calculate location in array that matches current boardNumber
-      //let filteredBoard = workBoard.map((currentElement, index) => ({ ...currentElement, key1: index })).filter((currentElement) => {
       let filteredBoard = workBoard.filter(currentElement => {
         return currentElement.boardNumber !== undefined && currentElement.invisible !== true;
       });
-      const newPosition1 = Math.min(prevGameState.position1 + randomNumber, filteredBoard.length);
+      let newPosition1 = prevGameState.position1;
+      let newPosition2 = prevGameState.position2;
+      if (prevGameState.whoseTurn === 1 || prevGameState.numberOfPlayers === 1) {
```

Comment:
Why do you need the number of players check here? How would whoseTurn ever be not 1, if numberOfPlayers is 1?

Reply:
Agreed....fixed.